### PR TITLE
[RFR] added custom rules and labels functionality + json data for eac…

### PR DESCRIPTION
**Pull request by stillalearner**
…h target + some tests

### CODE REVIEW

File: cypress/utils/utils.ts [hunk 1]
```diff
@@ -105,3 +111,26 @@ export function selectProject(projectName: string): void {
     click(chooseProject);
     clickByText("button", projectName);
 }
+
+export function getRandomApplicationData(jsonData): projectData {
```
Comment (sshveta) [inline cypress/utils/utils.ts:115]: jsonData is very generic . let's name it applicationData

File: cypress/fixtures/json/data.json [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    },
+
+    "Linux": {
+        "name": "Linux",
+
+        "apps": ["agent.jar"],
+        "targets": ["Linux"]
+    },
+
+    "Java2Jakarta": {
```
Comment (sshveta) [inline cypress/fixtures/json/data.json:56]: We do not need all of these at the moment. We will slowly test and add applications one by one . Let's start with what we have in MTA and check if storypoint match between MTA and MTR .

**Review by sshveta — CHANGES_REQUESTED** [review]

File: cypress/e2e/tests/all_targets_analysis.test.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        });
+
+        login();
+    });
+
+    let index = 0;
+
+    let targets = ["Multiple_eap8", "Containerization", "Linux", "Discovery"];
+
+    targets.forEach((target) => {
```
[the remark this answers is not in the document]
Comment (stillalearner) [reply cypress/e2e/tests/all_targets_analysis.test.ts:19]: done

File: cypress/fixtures/json/data.json (hunk 2 above)
Comment (stillalearner) [reply cypress/fixtures/json/data.json:56]: done

File: cypress/utils/utils.ts (hunk 1 above)
Comment (stillalearner) [reply cypress/utils/utils.ts]: done

**Review by sshveta — APPROVED** [review]

**Pull request merged**